bip-musig2.mediawiki

```diff
@@ -410,6 +391,49 @@ The reference implementation is for demonstration purposes only and not to be us
 
 == Remarks on Security and Correctness ==
 
+=== Modifications to Nonce Generation ===
+
+Implementors must avoid modifying the ''NonceGen'' algorithm without being fully aware of the implications.
+The following two modifications are secure when applied correctly and may be useful in special circumstances, e.g., in very restricted environments where secure randomness is not available.
+
+First, on systems where obtaining uniformly random values is much harder than maintaining a global atomic counter, it can be beneficial to modify ''NonceGen''.
+Instead of drawing ''rand' '' uniformly at random, ''rand' '' can be the value of an atomic counter that is incremented whenever it is read.
+With this modification, the secret signing key ''sk'' of the signer generating the nonce is '''not''' an optional argument and must be provided to ''NonceGen''.
+The security of the resulting scheme is then depending on the requirement that the counter must never return the same output in two ''NonceGen'' invocations with the same ''sk''.
+
+Second, if there is a unique signer who is supposed to send the ''pubnonce'' last, it is possible to modify nonce generation for this single signer to not require high-quality randomness.
+Such a nonce generation algorithm ''DeterministicSign'' is specified below.
+Note that there are no optional arguments except ''rand'', which can be ommitted if randomness is entirely unavailable.
+''DeterministicSign'' requires the argument ''othernonce'' which is the output of ''NonceAgg'' run on the ''pubnonce'' value of all other signers.
```

Comment:
```suggestion
''DeterministicSign'' requires the argument ''aggothernonce'' which is the output of ''NonceAgg'' run on the ''pubnonce'' value of all other signers.
```

It's slightly longer but much clearer, I think. Otherwise "othernonce" sounds to me like a third kind of nonce (beside pub and sec)


----

> `output of ''NonceAgg'' run on the ''pubnonce'' value of all other signers.`

This sounds like the caller is required to run NonceAgg but in fact it's still possible to trust an aggregator with it.

Reply:
Ok, fixed this.